File: VB/UIAutomationTestingExample/LogInForm.vb
Imports DevExpress.XtraEditors
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace UIAutomationTestingExample

    Public Partial Class LogInForm
        Inherits XtraForm

        Private userService As UserService = New UserService()

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Async Sub logInButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            logInButton.Enabled = False
            Dim errorText As String = Await userService.LogIn(userNameEditor.Text, passwordEditor.Text)
            If Not String.IsNullOrEmpty(errorText) Then
                errorLabel.Text = errorText
                errorLabel.Visible = True
                logInButton.Enabled = True
            Else
                errorLabel.Text = String.Empty
                errorLabel.Visible = False
                DialogResult = DialogResult.OK
                logInButton.Enabled = True
                Close()
            End If
        End Sub
    End Class

    Public Class UserService

        Private TestValidUsername As String = "TestExistingUser"

        Private TestValidPassword As String = "nJt34%77fytpl"

        Public Async Function LogIn(ByVal username As String, ByVal password As String) As Task(Of String)
            Await Task.Delay(2000) ' Sets the delay for demonstration purposes.
            If Equals(username, TestValidUsername) AndAlso Equals(password, TestValidPassword) Then
                Return String.Empty
            Else
                Return "Invalid User or Password"
            End If
        End Function
    End Class
End Namespace

File: CS/UIAutomationTestingExample/LogInForm.cs
using DevExpress.XtraEditors;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace UIAutomationTestingExample {
    public partial class LogInForm : DevExpress.XtraEditors.XtraForm {
        UserService userService = new UserService();
        public LogInForm() {
            InitializeComponent();
        }

        async void logInButton_Click(object sender, EventArgs e) {
            logInButton.Enabled = false;
            string errorText = await userService.LogIn(userNameEditor.Text, passwordEditor.Text);
            if (!string.IsNullOrEmpty(errorText)) {
                errorLabel.Text = errorText;
                errorLabel.Visible = true;
                logInButton.Enabled = true;
            }
            else {
                errorLabel.Text = string.Empty;
                errorLabel.Visible = false;
                DialogResult = DialogResult.OK;
                logInButton.Enabled = true;
                Close();
            }
        }
    }

    public class UserService {
        string TestValidUsername = "TestExistingUser";
        string TestValidPassword = "nJt34%77fytpl";
        public async Task<string> LogIn(string username, string password) {
            await Task.Delay(2000); // Sets the delay for demonstration purposes.
            if (username == TestValidUsername && password == TestValidPassword)
                return string.Empty;
            else
                return "Invalid User or Password";

        }
    }
}
